clients/src/main/java/org/apache/kafka/common/metrics/KafkaMetric.java

```diff
@@ -26,12 +28,23 @@
     private final Object lock;
     private final Time time;
     private final Measurable measurable;
+    private final Gauge<?> gauge;
```

Comment:
Thinking about this, another option is to introduce the super interface now, but with no methods. We would then do an `instanceof` check instead of having 2 fields. We would then replace all `Gauge` overloads with an overload that takes the superinterface.